### PR TITLE
go.mod: fix dependencies

### PR DESCRIPTION
**Pull request by zimbatm**
Running `go test -v ./...` on master currently fails with:

    go/pkg/mod/k8s.io/client-go@v11.0.0+incompatible/rest/request.go:598:31: not enough arguments in call to watch.NewStreamWatcher

Follow
https://github.com/kubernetes/client-go/blob/master/INSTALL.md#go-modules
and use the latest release of kubernetes instead. Fixed with:

    export GO111MODULE=on
    go get k8s.io/client-go@kubernetes-1.16.3
    go mod tidy

There are also failing tests that are not fixed by this PR.

### CODE REVIEW

**Comment by FxKu** [comment]
Isn't this duplicate to #705 ? :smiley: 
I've also updated deps in PR #478, which we've just merge. So should be fixed now.

**Comment by zimbatm** [comment]
The issue is that `go get sigs.k8s.io/kind@v0.5.1` in `e2e/Makefile` bumps the version of the go-client, which then breaks back-compatibility in the other code.

**Comment by zimbatm** [comment]
Closed #705 as it was outdated. This is a bit different as it fixes the build, whereas #705 was just trying to cleanup things.

File: go.mod [hunk 1]
```diff
@@ -4,20 +4,22 @@ go 1.12
...
+	golang.org/x/crypto v0.0.0-20191206172530-e9b2fee46413 // indirect
+	golang.org/x/net v0.0.0-20191209160850-c0dbc17a3553 // indirect
+	golang.org/x/sys v0.0.0-20191210023423-ac6580df4449 // indirect
+	golang.org/x/tools v0.0.0-20191209225234-22774f7dae43 // indirect
 	gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15 // indirect
 	gopkg.in/yaml.v2 v2.2.4
 	k8s.io/api v0.0.0-20191121015604-11707872ac1c
-	k8s.io/apiextensions-apiserver v0.0.0-20191121021419-88daf26ec3b8
-	k8s.io/apimachinery v0.0.0-20191121015412-41065c7a8c2a
-	k8s.io/client-go v11.0.0+incompatible
```
Comment (zimbatm) [inline go.mod]: this is the important line in the changeset

**Comment by sdudoladov** [comment]
:+1:

**Comment by FxKu** [comment]
:+1:

**Comment by FxKu** [comment]
Thanks @zimbatm for providing this fix. For some reason, I didn't ran into these issues. But, go-client version should not be raised by the kind client.

**Pull request merged**